src/components/Deposit.js

```diff
@@ -1,540 +1,536 @@
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    constant: false,
+    inputs: [{ internalType: "address", name: "_governance", type: "address" }],
+    name: "setGovernance",
+    outputs: [],
+    payable: false,
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    constant: true,
+    inputs: [],
+    name: "symbol",
+    outputs: [{ internalType: "string", name: "", type: "string" }],
+    payable: false,
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    constant: true,
+    inputs: [],
+    name: "totalSupply",
+    outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
+    payable: false,
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    constant: false,
+    inputs: [
+      { internalType: "address", name: "recipient", type: "address" },
+      { internalType: "uint256", name: "amount", type: "uint256" },
+    ],
+    name: "transfer",
+    outputs: [{ internalType: "bool", name: "", type: "bool" }],
+    payable: false,
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    constant: false,
+    inputs: [
+      { internalType: "address", name: "sender", type: "address" },
+      { internalType: "address", name: "recipient", type: "address" },
+      { internalType: "uint256", name: "amount", type: "uint256" },
+    ],
+    name: "transferFrom",
+    outputs: [{ internalType: "bool", name: "", type: "bool" }],
+    payable: false,
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+const usdtAddress = "0xFbac52229E7Ca8A90bE9403E62891185Cf3b2D47";
 
+const Deposit = {
+  depositUSDT: async (depositAmount) => {
     const usdtContract = new ethers.Contract(usdtAddress, usdtAbi, signer);
     const contract = new ethers.Contract(contractAddress, contractAbi, signer);
-    
+
     const approveAmount = ethers.utils.parseUnits("1000", 18); // Set the amount you want to approve
     const approveTx = await usdtContract.approve(
       contractAddress,
       approveAmount
     );
-    alert("Approve successful! Please wait to deposit...")
+    alert("Approve successful! Please wait to deposit...");
     await approveTx.wait();
-    
 
     // Deposit the approved USDT tokens
     // const depositAmount_acc = ethers.utils.parseUnits("10", 18);
     const depositAmount_acc = ethers.utils.parseUnits(depositAmount, 18); // Set the deposit amount
     // console.log(typeof "10", typeof depositAmount, depositAmount)
     const depositTx = await contract.deposit(depositAmount_acc);
     await depositTx.wait();
+  },
 
+  getUserWalletBalance: async () => {
+    const usdtContract = new ethers.Contract(usdtAddress, usdtAbi, signer);
+    try {
+      const userAddress = await signer.getAddress();
+      const balance = await usdtContract.balanceOf(userAddress);
+
+      // USDT has 6 decimals, so you may need to divide the balance by 1e6
+      const formattedBalance = ethers.utils.formatUnits(balance, 18);
```

Comment:
The comment about USDT having 6 decimals is noted, but the code uses 18 decimals when calling ethers.utils.formatUnits.

Reply:
This comment is from code provided by Danny, USDT actually has 18 decimal after I tried for multiple times, I will correct the wrong comment.

Reply:
Corrected.